driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -519,12 +523,10 @@ public String toString() {
 
     public static class Factory {
 
-        private final ExecutorService bossExecutor = Executors.newCachedThreadPool();
-        private final ExecutorService workerExecutor = Executors.newCachedThreadPool();
         public final HashedWheelTimer timer = new HashedWheelTimer(new ThreadFactoryBuilder().setNameFormat("Timeouter-%d").build());
 
-        private final ChannelFactory channelFactory = new NioClientSocketChannelFactory(bossExecutor, workerExecutor);
-        private final ChannelGroup allChannels = new DefaultChannelGroup();
+        private final EventLoopGroup eventLoopGroup = new NioEventLoopGroup();
```

Comment:
It looks like NioEventLoopGroup you can pass in a thread factory (new NioEventLoopGroup(0, threadFactory).  Where previously you couldn't override the thread names of "New I/O Worker X".  Wondering if we could provide our own named thread factory (for [JAVA-583](https://datastax-oss.atlassian.net/browse/JAVA-583)), which would differentiate Netty IO workers created by the driver from those created by other libraries used within an application.  Mind if I add this?